packages/xstate-fsm/src/index.ts

```diff
@@ -87,27 +87,32 @@ export function createMachine<
   } = {}
 ): StateMachine.Machine<TContext, TEvent, TState> {
   if (!IS_PRODUCTION) {
-    Object
-      .keys(fsmConfig['states'])
-      .forEach((state) => {
-        if (fsmConfig['states'][state].states) {
-          throw new Error(`Nested finite states not supported. 
+    Object.keys(fsmConfig.states).forEach((state) => {
+      if (fsmConfig.states[state].states) {
+        throw new Error(`Nested finite states not supported.
             Please check the configuration for the "${state}" state.`);
-        }
-      });
+      }
+    });
   }
-  
+
+  const getStateByValue = (
```

Comment:
This can probably be moved to a `getStateByValue(machine, state)` function, what do you think? Will reduce output code size.

Reply:
This one is tricky to make. We need properties that `Machine` do not expose:
- `actions`
- `context`

I tried to use currying to workaround the issue, but it increases the size:

```
function createGetStateByValue<
  TContext extends object,
  TEvent extends EventObject,
  TState extends Typestate<TContext>
  >(
  machineConfig: StateMachine.Config<TContext, TEvent, TState>,
  actions: StateMachine.ActionMap<TContext, TEvent> = {},
): (state: TState['value']) => StateMachine.State<TContext, TEvent, TState> {
  return (state: TState['value']) => ({
    value: state,
    actions: toArray(machineConfig.states[state].entry).map((action) =>
      toActionObject(action, actions)
    ),
    context: machineConfig.context!,
    matches: createMatcher(state)
  });
}
```